plugins/templates/rules/xadatasources.j2

```diff
@@ -0,0 +1,23 @@
+{ "xadatasource" => {
+  "{{ values['name'] }}" => {
+    {% if 'pool_name' in values %} "pool-name" => "{{ values['pool_name'] }}", {% endif %}
+    "xa-datasource-properties" => {
+      {% if 'url' in values['xa-datasource_properties'] %} "URL" => { "value" => "{{ values['xa_datasource_properties']['url'] }}" } {% endif %}
```

Comment:
only a typo here (`xa-datasource_properties` -> `xa_datasource_properties`), rest is LGTM!

Reply:
You got sharp eyes :)
I'll fix it right away